Bearded.Graphics/Debugging/KHRDebugExtension.cs

```diff
@@ -0,0 +1,57 @@
+using System.Linq;
+using OpenTK.Graphics.OpenGL;
+
+namespace Bearded.Graphics.Debugging
+{
+    public abstract class KHRDebugExtension
+    {
+        private static KHRDebugExtension? instance;
+        public static KHRDebugExtension Instance => instance ??= createInstanceWithAvailableFunctionality();
+
+        private KHRDebugExtension()
+        {
+        }
+
+        private static KHRDebugExtension createInstanceWithAvailableFunctionality()
+        {
+            if (khrDebugExtensionIsAvailable())
+                return new ActualImplementation();
+            return new DummyImplementation();
+        }
+
+        private static bool khrDebugExtensionIsAvailable()
```

Comment:
nit: should this be a property?

Reply:
It does a lot of work for being a property I think.